arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,46 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const unsigned long PRINT_INTERVAL = 100;
+unsigned long previousPrintout     = 0;
+
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+GY50 gyroscope(arduinoRuntime, 37);
+
+const auto pulsesPerMeter = 600;
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 400;
+
+
+DirectionalOdometer leftOdometer{arduinoRuntime, smartcarlib::pins::v2::leftOdometerPins, []() { leftOdometer.update(); }, pulsesPerMeter};
+DirectionalOdometer rightOdometer{arduinoRuntime, smartcarlib::pins::v2::leftOdometerPins, []() { leftOdometer.update(); }, pulsesPerMeter};
+
+
+
+SmartCar car(arduinoRuntime, control, gyroscope, leftOdometer, rightOdometer);
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+void setup()
+{
+    Serial.begin(9600);
+    car.enableCruiseControl();
+    car.setSpeed(0.5); // Maintain a speed of 1.5 m/sec
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+   // Maintain the speed and update the heading
+    car.update();
+    // Stop after moving 1 meter
+    if (front.getDistance() != 0 && front.getDistance()< 200)
```

Comment:
You are conducting **two** measurements that may well return different results. If for example the first call return `300` and the second one `0` you will stop the car even though the obstacle would be very far away